main.go

```diff
@@ -2,107 +2,26 @@ package main
 
 import (
 	"context"
-	"database/sql"
 	"flag"
-	"fmt"
 	"time"
 
 	"github.com/cyverse-de/configurate"
 	"github.com/cyverse-de/go-mod/logging"
+	"github.com/cyverse-de/go-mod/otelutils"
 	"github.com/cyverse-de/go-mod/protobufjson"
-	"github.com/cyverse-de/p/go/svcerror"
-	"github.com/cyverse-de/p/go/user"
-	"github.com/doug-martin/goqu/v9"
 	"github.com/jmoiron/sqlx"
 	_ "github.com/lib/pq"
 	"github.com/nats-io/nats.go"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/viper"
-	"google.golang.org/protobuf/types/known/timestamppb"
+	"github.com/uptrace/opentelemetry-go-extra/otelsql"
+	"github.com/uptrace/opentelemetry-go-extra/otelsqlx"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
 )
 
 var log = logging.Log.WithFields(logrus.Fields{"service": "discoenv-users-service"})
 
-type lookupUser struct {
-	Username        string `db:"username"`
-	UserID          string `db:"id"`
-	LoginCount      uint32 `db:"login_count"`
-	Preferences     string `db:"preferences"`
-	PreferencesID   string `db:"preferences_id"`
-	Logins          []lookupLogin
-	SavedSearches   string `db:"saved_searches"`
-	SavedSearchesID string `db:"saved_searches_id"`
-}
-
-type lookupLogin struct {
-	IPAddress  sql.NullString `db:"ip_address"`
-	UserAgent  sql.NullString `db:"user_agent"`
-	LoginTime  sql.NullTime   `db:"login_time"`
-	LogoutTime sql.NullTime   `db:"logout_time"`
-}
-
-func lookupLogins(dbconn *sqlx.DB, userID string, limit, offset uint) ([]lookupLogin, error) {
-	var err error
-
-	usersT := goqu.T("users")
-	loginsT := goqu.T("logins")
-	loginsQ := goqu.From(loginsT).
-		Join(usersT, goqu.On(loginsT.Col("user_id").Eq(usersT.Col("id")))).
-		Where(usersT.Col("id").Eq(userID)).
-		Select(
-			loginsT.Col("ip_address"),
-			loginsT.Col("user_agent"),
-			loginsT.Col("login_time"),
-			loginsT.Col("logout_time"),
-		).
-		Order(loginsT.Col("login_time").Desc()).
-		Limit(limit).
-		Offset(offset)
-
-	loginsQueryString, _, err := loginsQ.ToSQL()
-	if err != nil {
-		return nil, err
-	}
-
-	rows, err := dbconn.QueryxContext(context.Background(), loginsQueryString)
-	if err != nil {
-		return nil, err
-	}
-
-	logins := []lookupLogin{}
-
-	for rows.Next() {
-		login := lookupLogin{}
-		if err = rows.StructScan(&login); err != nil {
-			return nil, err
-		}
-
-		logins = append(logins, login)
-	}
-
-	return logins, nil
-}
-
-func loginCount(dbconn *sqlx.DB, userID string) (uint, error) {
-	var err error
-
-	usersT := goqu.T("users")
-	loginsT := goqu.T("logins")
-	countQ := goqu.From(loginsT).
-		Join(usersT, goqu.On(loginsT.Col("user_id").Eq(usersT.Col("id")))).
-		Where(usersT.Col("id").Eq(userID)).
-		Select(goqu.COUNT("*"))
-	q, _, err := countQ.ToSQL()
-	if err != nil {
-		return 0, err
-	}
-
-	var count uint
-	if err = dbconn.QueryRowxContext(context.Background(), q).Scan(&count); err != nil {
-		return count, err
-	}
-	return count, nil
-}
+const serviceName = "discoenv-users"
```

Comment:
You can do as you see fit, of course, but in other services what I've usually been doing is putting the `const serviceName` right at the top, and then using the same constant for stuff like the `service` field in logging. Just seems cleaner. I'd imagine we should remove the `-service` suffix in the log field for consistency with our other services anyway (not that we use the logs much right now)

Reply:
lol, this pulled in a gigantic diff, whoops. I'm really only commenting on the `var log` line and the `const serviceName` line

Reply:
No worries, and thanks for the suggestion. Should be done now.